node/lib/util/git_util.js

```diff
@@ -131,16 +134,42 @@ exports.getTrackingInfo = co.wrap(function *(branch) {
     }
     const name = upstream.shorthand();
     const parts = name.split("/");
+    const config = yield repo.config();
+
+    // Try to read a 'pushRemote' for the branch.
+
+    let pushRemote = null;
+    try {
+        pushRemote = yield config.getStringBuf(
+                                    `branch.${branch.shorthand()}.pushRemote`);
+    }
+    catch (e) {
+        // Unfortunately, no other way to handle a missing config entry
```

Comment:
I feel like we do this a lot -- could we extract a method?

Reply:
OK, see if this is better.